packages/backend-common/src/reading/AwsS3UrlReader.ts

```diff
@@ -156,39 +157,95 @@ export class AwsS3UrlReader implements UrlReader {
    * If accessKeyId and secretAccessKey are missing, the standard credentials provider chain will be used:
    * https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html
    */
-  private static buildCredentials(
+  private static buildStaticCredentials(
+    accessKeyId: string,
+    secretAccessKey: string,
+  ): AwsCredentialIdentityProvider {
+    return async () => {
+      return Promise.resolve({
+        accessKeyId,
+        secretAccessKey,
+      });
+    };
+  }
+
+  private static async buildCredentials(
+    credsManager: AwsCredentialsManager,
+    region: string,
     integration?: AwsS3Integration,
-  ): aws.Credentials | CredentialsOptions | undefined {
+  ): Promise<AwsCredentialIdentityProvider> {
+    // Fall back to the default credential chain if neither account ID
+    // nor explicit credentials are provided
     if (!integration) {
-      return undefined;
+      return (await credsManager.getCredentialProvider()).sdkCredentialProvider;
     }
 
+    // Pull credentials from the techdocs config section (deprecated)
```

Comment:
this isn't the techdocs config though right? it's under `integrations` right

Reply:
That's correct, and I removed the misleading comment.